testing/dart/isolate_name_server_test.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:isolate';
+import 'dart:ui';
+
+import 'package:test/test.dart';
+
+const kPortName = 'foobar';
+const kErrorCode = -1;
+const kStartCode = 0;
+const kCloseCode = 1;
+const kDeletedCode = 2;
+
+void isolateSpawnEntrypoint(SendPort port) {
+  sendHelper(int code, [String message = '']) {
+    port.send(<dynamic>[code, message]);
+  }
+
+  SendPort shared = IsolateNameServer.lookupPortByName(kPortName);
+  if (shared == null) {
+    sendHelper(kErrorCode, 'Could not find port: $kPortName');
+    return;
+  }
+
+  // ack that the SendPort lookup was successful.
+  sendHelper(kStartCode);
+
+  //shared.send(kPortName);
+  sendHelper(kCloseCode);
+
+  // We'll fail if the ReceivePort's callback is called more than once. Try to
+  // send another message to ensure we don't crash.
+  shared.send('garbage');
+
+  bool result = IsolateNameServer.removePortNameMapping(kPortName);
+  if (result) {
+    sendHelper(kDeletedCode);
+  } else {
+    sendHelper(kErrorCode, 'Was unable to remove mapping for $kPortName');
+  }
+}
+
+void main() {
+  tearDown(() {
+    IsolateNameServer.removePortNameMapping(kPortName);
+  });
+
+  test('simple isolate name server', () {
+    // Mapping for 'foobar' isn't set. Check these cases to ensure correct
+    // negative response.
+    expect(IsolateNameServer.lookupPortByName(kPortName), isNull);
+    expect(IsolateNameServer.removePortNameMapping(kPortName), isFalse);
+
+    // Register a SendPort.
+    final receivePort = new ReceivePort();
```

Comment:
Do you need to close these ports?

Reply:
They're not being used for sending + receiving so I _think_ this should be fine.